src/pages/[platform]/build-a-backend/data/working-with-files/index.mdx

```diff
@@ -210,13 +254,28 @@ if (!updatedSong?.coverArtPath) return;
 
 // Retrieve the file's signed URL:
 const signedURL = await getUrl({ path: updatedSong.coverArtPath });
-
 ```
 
+</InlineFilter>
+
 ## Query a record and retrieve the associated file
 
 To retrieve the file associated with a record, first query the record, then use Storage to get the signed URL. The signed URL can then be used to download the file, display an image, etc:
+<InlineFilter filters={["swift"]}>
+```swift title="ContentView"
+// Get the song record
+let song = try await Amplify.API.query(request: .get(Song.self, byId: currentSong.id)).get()
+
+// Download the art cover
+let imageData = try await Amplify.Storage.downloadData(path: .fromString((song?.coverArtPath)!)).value
```

Comment:
should we guard instead of force unwrapping here? the full example and some examples below still use guard, so guarding here will create consistency. also force unwrap could crash and usually we want to avoid that.